tiledb/fragment.py

```diff
@@ -12,6 +12,86 @@
+    :param to_vacuum_num: Number of already consolidated fragments to vacuum
+    :param to_vacuum_uri: URIs of already consolidated fragments to vacuum
+
+    **Example:**
+
+    >>> import tiledb, numpy as np, tempfile
+    >>> with tempfile.TemporaryDirectory() as tmp:
+    ...     # The array will be 4x4 with dimensions "rows" and "cols", with domain [1,4] and space tiles 2x2
+    ...     dom = tiledb.Domain(
+    ...         tiledb.Dim(name="rows", domain=(1, 4), tile=2, dtype=np.int32),
+    ...         tiledb.Dim(name="cols", domain=(1, 4), tile=2, dtype=np.int32),
+    ...     )
+    ...     # The array will be dense with a single attribute "a" so each (i,j) cell can store an integer.
+    ...     schema = tiledb.ArraySchema(
+    ...         domain=dom, sparse=False, attrs=[tiledb.Attr(name="a", dtype=np.int32)]
+    ...     )
+    ...     # Create the (empty) array on disk.
+    ...     tiledb.Array.create(tmp + "/array", schema)
+    ...
+    ...     # Write three fragments to the array
+    ...     with tiledb.DenseArray(array_name, mode="w") as A:
+    ...         A[1:3, 1:5] = np.array(([1, 2, 3, 4, 5, 6, 7, 8]))
+    ...     with tiledb.DenseArray(array_name, mode="w") as A:
+    ...         A[2:4, 2:4] = np.array(([101, 102, 103, 104]))
+    ...     with tiledb.DenseArray(array_name, mode="w") as A:
+    ...         A[3:4, 4:5] = np.array(([202]))
+    ...
+    ...     # tiledb.array_fragments() requires TileDB-Py version > 0.8.5
+    ...     fragments_info = tiledb.array_fragments(array_name)
+    ...
+    ...     "====== FRAGMENTS  INFO ======"
+    ...     fragments_info.array_uri
+    ...     f"number of fragments: {len(fragments_info)}"
+    ...     f"nonempty domains: {fragments_info.nonempty_domain}"
+    ...     f"sparse fragments: {fragments_info.sparse}"
+    ...
+    ...     for fragment_num, fragment in enumerate(fragments_info, start=1):
+    ...         f"===== FRAGMENT NUMBER {fragment.num} ====="
+    ...         f"is dense: {fragment.dense}"
+    ...         f"cell num: {fragment.cell_num}"
+    ...         f"has consolidated metadata: {fragment.has_consolidated_metadata}"
+    ...         f"nonempty domain: {fragment.nonempty_domain}"
+    ...         f"timestamp range: {fragment.timestamp_range}"
+    ...         f"fragment version: {fragment.version}"
+    ====== FRAGMENTS  INFO ======
+    array uri: fragment_info
+    number of fragments: 3
+    nonempty domains: (((1, 2), (1, 4)), ((2, 3), (2, 3)), ((3, 3), (4, 4)))
+    sparse fragments: (False, False, False)
+    ===== FRAGMENT NUMBER 0 =====
+    is dense: True
+    cell num: 8
+    has consolidated metadata: False
+    nonempty domain: ((1, 2), (1, 4))
+    timestamp range: (1628273555397, 1628273555397)
+    version: 9
+    ===== FRAGMENT NUMBER 1 =====
+    is dense: True
+    cell num: 16
+    has consolidated metadata: False
+    nonempty domain: ((2, 3), (2, 3))
+    timestamp range: (1628273555409, 1628273555409)
```

Comment:
Shouldn't this docstring test fail because the timestamps will be different every time it is run?

Also we might need to do this at the bottom to run as doctest?: https://stackoverflow.com/a/2708239